cirq-web/cirq_ts/src/bloch_sphere/components/meridians_test.ts

```diff
@@ -0,0 +1,189 @@
+      const expectedLineNumbers = [4, 0, 18, 299, 4];
+      lineValues.forEach((el, index) => {
+        const meridians = new Meridians(
+          DEFAULT_RADIUS,
+          el,
+          Orientation.HORIZONTAL
+        );
+        expect(meridians.children.length).to.equal(expectedLineNumbers[index]);
+      });
+    });
+
+    it('createVerticalMeridians() generates the correct number of lines given valid numbers', () => {
+      const lineValues = [4, 0, 17, 299, 4.123];
+      const expectedLineNumbers = [4, 0, 18, 299, 4];
+      lineValues.forEach((el, index) => {
+        const meridians = new Meridians(
+          DEFAULT_RADIUS,
+          el,
+          Orientation.VERTICAL
+        );
+        expect(meridians.children.length).to.equal(expectedLineNumbers[index]);
+      });
+    });
+
+    it('createHorizontalChordMeridians() defaults if given invalid inputs (-1, 301)', () => {
+      const lineValues = [-1, 301];
+      const expectedLineNumbers = [7, 7];
+      lineValues.forEach((el, index) => {
+        const meridians = new Meridians(
+          DEFAULT_RADIUS,
+          el,
+          Orientation.HORIZONTAL_CHORD
+        );
+        expect(meridians.children.length).to.equal(expectedLineNumbers[index]);
+      });
+    });
+
+    it('createHorizontalCircleMeridians() defaults if given invalid inputs (-1, 301)', () => {
+      const lineValues = [-1, 301];
+      const expectedLineNumbers = [4, 4];
+      lineValues.forEach((el, index) => {
+        const meridians = new Meridians(
+          DEFAULT_RADIUS,
+          el,
+          Orientation.HORIZONTAL
+        );
+        expect(meridians.children.length).to.equal(expectedLineNumbers[index]);
+      });
+    });
+
+    it('createVerticalMeridians() defaults if given invalid inputs (-1, 301)', () => {
+      const lineValues = [-1, 301];
+      const expectedLineNumbers = [4, 4];
+      lineValues.forEach((el, index) => {
+        const meridians = new Meridians(
+          DEFAULT_RADIUS,
+          el,
+          Orientation.VERTICAL
+        );
+        expect(meridians.children.length).to.equal(expectedLineNumbers[index]);
+      });
+    });
+
+    it('changing the number of horizontal chord meridians changes the positions correctly with scale', () => {
+      const meridians = new Meridians(
+        DEFAULT_RADIUS,
+        9,
+        Orientation.HORIZONTAL_CHORD
+      );
+
+      const positions = [
+        new Vector3(0, 0, 0),
+        new Vector3(0, 4.5, 0),
+        new Vector3(0, -4.5, 0),
+        new Vector3(0, 3.375, 0),
+        new Vector3(0, -3.375, 0),
+        new Vector3(0, 2.25, 0),
+        new Vector3(0, -2.25, 0),
+        new Vector3(0, 1.125, 0),
+        new Vector3(0, -1.125, 0),
+      ];
+
+      meridians.children.forEach((el, index) => {
+        expect(el.position).to.eql(positions[index]);
+      });
+    });
+
+    it('giving bad orientation enum in the constructor will return no circles', () => {
```

Comment:
We should throw an error here instead of silently creating an empty object. The principle is "Fail fast" - let the user know that something wrong as early as possible, with a meaningful error message. Otherwise they'll have to run the whole thing, display it in a browser and then start debugging.